### PR TITLE
Improve authentication methods

### PR DESCRIPTION
**Pull request by bjoernricks**
**What**:

Refactor auth methods module to reflect usage in our frontend.

**Why**:

The more generic API of GMP with method names, keys and values is very unclear and doesn't specifiy the exact usage for radius and ldap. This has to be looked up in the implementation at the moment. With the new API it is obvious which arguments are expected and which data is returned.

**How**:

**Checklist**:

- [x] Tests
- [x] [CHANGELOG](https://github.com/greenbone/hyperion/blob/master/CHANGELOG.md) Entry
- [ ] Documentation

### CODE REVIEW

File: selene/schema/authentication_methods/mutations.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+import graphene
+
+from selene.schema.utils import (
+    get_gmp,
+    require_authentication,
+)
+
+
+def _to_bool(value: bool) -> str:
```
Comment (bjoernricks) [inline selene/schema/authentication_methods/mutations.py:27]:
```suggestion
def _to_bool(value: str) -> bool:
```

File: selene/schema/authentication_methods/queries.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+)
+
+
+class CaCertificate(CACertificateMixin, graphene.ObjectType):
+    certificate = graphene.String(
+        description='Base64 encoded data of the CA certificate'
+    )
+    time_status = graphene.String(
+        description=(
+            'Whether the certificate is valid, ' 'expired or not active yet.'
```
Comment (bjoernricks) [inline selene/schema/authentication_methods/queries.py:40]:
```suggestion
            'Whether the certificate is valid, expired or not active yet.'
```

File: selene/tests/authentication_methods/test_describe_auth.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+                ldapAuthenticationSettings {
+                    authDn
+                }
+            }
+            '''
+        )
+
+        self.assertResponseAuthenticationRequired(response)
+
+    def test_describe_auth_ldap(self, mock_gmp: GmpMockFactory):
```
Comment (bjoernricks) [inline selene/tests/authentication_methods/test_describe_auth.py:45]:
```suggestion
    def test_ldap_authentication_settings(self, mock_gmp: GmpMockFactory):
```

File: selene/tests/authentication_methods/test_describe_auth.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        self.assertEqual(
+            ca_certificate['expirationTime'], '2026-03-31T13:56:33+00:00'
+        )
+        self.assertEqual(
+            ca_certificate['certificate'],
+            '-----BEGIN CERTIFICATE-----MIIDazCCAlOgA-----END CERTIFICATE-----',
+        )
+        self.assertEqual(ca_certificate['timeStatus'], 'valid')
+
+    def test_describe_auth_radius(self, mock_gmp: GmpMockFactory):
```
Comment (bjoernricks) [inline selene/tests/authentication_methods/test_describe_auth.py:107]:
```suggestion
    def test_radius_authentication_settings(self, mock_gmp: GmpMockFactory):
```

File: selene/schema/authentication_methods/queries.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+from selene.schema.utils import (
+    get_datetime_from_element,
+    get_gmp,
+    get_text_from_element,
+    require_authentication,
+    XmlElement,
+)
+
+
+class CaCertificate(CACertificateMixin, graphene.ObjectType):
```
Comment (sarahd93) [inline selene/schema/authentication_methods/queries.py:34]: This file should only contain the queries. CaCertificate, LDAPAuthenticationSettings and RADIUSAuthenticationSettings should be in fields.py. I think fields.py still contains the old fields right now.

**Review by sarahd93 — CHANGES_REQUESTED** [review]

File: selene/schema/authentication_methods/queries.py (hunk 5 above)
Comment (bjoernricks) [reply selene/schema/authentication_methods/queries.py:34]: @saberlynx could you care about that?

**Review by sarahd93 — APPROVED** [review]

**Pull request merged**